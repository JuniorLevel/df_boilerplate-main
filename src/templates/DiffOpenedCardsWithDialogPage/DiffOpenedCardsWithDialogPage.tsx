import React, { FC } from 'react';
import { Outlet } from '@tanstack/react-router';
import { sidebarButtons, data, filters } from './data';
import { OrdersFilterProviderPrmt } from '@/primitives/OrdersFilterPrmt/OrdersFilterContextPrmt';
import { LayoutPrmt } from '@/primitives/LayoutPrmt/LayoutPrmt';
import { FooterPrmt } from '@/primitives/FooterPrmt/FooterPrmt';
import { OrdersFilterOpenedPrmt } from '@/primitives/OrdersFilterPrmt/OrdersFilterOpenedPrmt/OrdersFilterOpenedPrmt';
import { OrderListPreviewContainerPrmt } from '@/primitives/OrderListPreviewPrmt/OrderListPreviewContainerPrmt/OrderListPreviewContainerPrmt';
import { OrderListDiffContainerPrmt } from '@/primitives/OrderListPreviewPrmt/OrderListDiffContainerPrmt/OrderListDiffContainerPrmt';
import { OrderListItemPrmt } from '@/primitives/OrderListPreviewPrmt/OrderListContainerPrmt/OrderListItemPrmt/OrderListItemPrmt';
import { OrderPrmt } from '@/primitives/OrderListPreviewPrmt/OrderPrmt/OrderPrmt';
import { RowPrmt } from '@/primitives/RowPrmt/RowPrmt';
import { ColPrmt } from '@/primitives/ColPrmt/ColPrmt';
import { PaginationPrmt as OrdersPaginationPrmt } from '@/primitives/PaginationPrmt/PaginationPrmt';

export const DiffOpenedCardsWithDialogPage: FC = () => (
	<OrdersFilterProviderPrmt>
		<LayoutPrmt buttons={sidebarButtons} title="Caption" avatar={{ title: 'AV' }} footer={<FooterPrmt />}>
			<RowPrmt gutter={[16, 16]}>
				<ColPrmt span={12}>
					<OrdersFilterOpenedPrmt open dialog filters={filters} />
					<OrderListPreviewContainerPrmt>
						<ColPrmt span={12}>
							<OrderListDiffContainerPrmt>
								<OrderListItemPrmt key={data[0].id} data={data[0]} />
								<OrderListItemPrmt key={data[1].id} data={data[1]} />
								<OrderListItemPrmt key={data[2].id} data={data[2]} />
								<OrderListItemPrmt key={data[3].id} data={data[3]} />
								<OrderListItemPrmt key={data[4].id} data={data[4]} />
								<OrderListItemPrmt key={data[5].id} data={data[5]} />
								<OrderListItemPrmt key={data[6].id} data={null} />
							</OrderListDiffContainerPrmt>
							<OrdersPaginationPrmt defaultCurrent={1} total={50} />
						</ColPrmt>
						<ColPrmt span={12}>
							<OrderPrmt />
						</ColPrmt>
					</OrderListPreviewContainerPrmt>
				</ColPrmt>
				<ColPrmt span={12}>
					<OrdersFilterOpenedPrmt open dialog filters={filters} />
					<OrderListPreviewContainerPrmt>
						<ColPrmt span={12}>
							<OrderListDiffContainerPrmt>
								<OrderListItemPrmt key={data[0].id} data={data[0]} />
								<OrderListItemPrmt key={data[1].id} data={data[1]} />
								<OrderListItemPrmt key={data[2].id} data={data[2]} />
								<OrderListItemPrmt key={data[3].id} data={data[3]} />
								<OrderListItemPrmt key={data[4].id} data={data[4]} />
								<OrderListItemPrmt key={data[5].id} data={data[5]} />
								<OrderListItemPrmt key={data[6].id} data={null} />
							</OrderListDiffContainerPrmt>
							<OrdersPaginationPrmt defaultCurrent={1} total={50} />
						</ColPrmt>
						<ColPrmt span={12}>
							<OrderPrmt />
						</ColPrmt>
					</OrderListPreviewContainerPrmt>
				</ColPrmt>
			</RowPrmt>
		</LayoutPrmt>
		<Outlet />
	</OrdersFilterProviderPrmt>
);
