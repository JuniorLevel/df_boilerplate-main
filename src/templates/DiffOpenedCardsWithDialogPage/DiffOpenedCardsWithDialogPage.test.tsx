import React from 'react';
import { describe, test } from '@jest/globals';
import { Context as ResponsiveContext } from 'react-responsive';
import { render, waitFor, act, cleanup } from '@testing-library/react';
import '../../../__mocks__/fileMock';
import { DiffOpenedCardsWithDialogPage } from './DiffOpenedCardsWithDialogPage';

describe('DiffOpenedCardsWithDialogPage', () => {
	afterEach(() => {
		cleanup();
	});
	test('should render DiffOpenedCardsWithDialogPage component without crashing', async () => {
		await act(async () => {
			const { container } = render(<DiffOpenedCardsWithDialogPage />);
			await waitFor(() => {
				expect(container).toBeInTheDocument();
			});
		});
	});

	test('matches the snapshot desktop', async () => {
		await act(async () => {
			const { container: desktop } = render(
				<ResponsiveContext.Provider value={{ width: 1920 }}>
					<DiffOpenedCardsWithDialogPage />
				</ResponsiveContext.Provider>
			);
			await waitFor(() => {
				expect(desktop).toMatchSnapshot();
			});
		});
	});

	test('matches the snapshot laptop', async () => {
		await act(async () => {
			const { container: laptop } = render(
				<ResponsiveContext.Provider value={{ width: 1024 }}>
					<DiffOpenedCardsWithDialogPage />
				</ResponsiveContext.Provider>
			);
			await waitFor(() => {
				expect(laptop).toMatchSnapshot();
			});
		});
	});

	test('matches the snapshot mobile', async () => {
		await act(async () => {
			const { container: mobile } = render(
				<ResponsiveContext.Provider value={{ width: 360 }}>
					<DiffOpenedCardsWithDialogPage />
				</ResponsiveContext.Provider>
			);
			await waitFor(() => {
				expect(mobile).toMatchSnapshot();
			});
		});
	});
});
