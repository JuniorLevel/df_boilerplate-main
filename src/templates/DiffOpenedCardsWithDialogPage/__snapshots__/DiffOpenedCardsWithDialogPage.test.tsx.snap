// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffOpenedCardsWithDialogPage matches the snapshot desktop 1`] = `<div />`;

exports[`DiffOpenedCardsWithDialogPage matches the snapshot laptop 1`] = `<div />`;

exports[`DiffOpenedCardsWithDialogPage matches the snapshot mobile 1`] = `<div />`;
